ext/js/pages/settings/anki-deck-generator-controller.js

```diff
@@ -0,0 +1,584 @@
+        const text = /** @type {HTMLInputElement} */ (this._renderTextInput).value;
+        let result;
+        try {
+            const noteData = await this._generateNoteData(text, mode, false);
+            result = noteData ? this._fieldsToTSV(noteData.fields) : `No definition found for ${text}`;
+        } catch (e) {
+            allErrors.push(toError(e));
+        }
+
+        /**
+         * @param {Error} e
+         * @returns {string}
+         */
+        const errorToMessageString = (e) => {
+            if (e instanceof ExtensionError) {
+                const v = e.data;
+                if (typeof v === 'object' && v !== null) {
+                    const v2 = /** @type {import('core').UnknownObject} */ (v).error;
+                    if (v2 instanceof Error) {
+                        return v2.message;
+                    }
+                }
+            }
+            return e.message;
+        };
+
+        const hasError = allErrors.length > 0;
+        infoNode.hidden = !(showSuccessResult || hasError);
+        if (hasError || !result) {
+            infoNode.textContent = allErrors.map(errorToMessageString).join('\n');
+        } else {
+            infoNode.textContent = showSuccessResult ? result : '';
+        }
+        infoNode.classList.toggle('text-danger', hasError);
+    }
+
+    /**
+     * @param {string} word
+     * @param {import('anki-templates-internal').CreateModeNoTest} mode
+     * @param {boolean} addMedia
+     * @returns {Promise<?import('anki.js').Note>}
+     */
+    async _generateNoteData(word, mode, addMedia) {
+        const optionsContext = this._settingsController.getOptionsContext();
+        const data = await this._getDictionaryEntry(word, optionsContext);
+        if (data === null) {
+            return null;
+        }
+        const {dictionaryEntry, text: sentenceText} = data;
+        const options = await this._settingsController.getOptions();
+        const context = {
+            url: window.location.href,
+            sentence: {
+                text: sentenceText,
+                offset: 0
+            },
+            documentTitle: document.title,
+            query: sentenceText,
+            fullQuery: sentenceText
+        };
+        const template = this._getAnkiTemplate(options);
+        const deckOptionsFields = options.anki.terms.fields;
+        const {general: {resultOutputMode, glossaryLayoutMode, compactTags}} = options;
+        const fields = [];
+        for (const deckField in deckOptionsFields) {
+            if (Object.prototype.hasOwnProperty.call(deckOptionsFields, deckField)) {
+                fields.push([deckField, deckOptionsFields[deckField]]);
+            }
+        }
+        const mediaOptions = addMedia ? {audio: {sources: options.audio.sources, preferredAudioIndex: null, idleTimeout: (Number.isFinite(options.anki.downloadTimeout) && options.anki.downloadTimeout > 0 ? options.anki.downloadTimeout : null)}} : null;
```

Comment:
Am in awe at the size of this line. Break it up into explanatory variables?

Reply:
Pulled the idleTimeout part out into a separate variable